toolchain/glibc.py

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env python
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+# Copyright: 2016 IBM
+# Author: Santhosh G <santhog4@linux.vnet.ibm.com>
+
+import os
+from avocado import Test
+from avocado import main
+from avocado.utils import process
+from avocado.utils import build
+from avocado.utils import archive
+from avocado.utils.software_manager import SoftwareManager
+from avocado.utils import distro
+from avocado.core import data_dir
+
+
+class glibc(Test):
+    def setUp(self):
+        sm = SoftwareManager()
+        detected_distro = distro.detect()
+        deps = ['gcc', 'make', 'gawk']
+        self.tmpdir = data_dir.get_tmp_dir()
+        self.build_dir = self.params.get('build_dir', default=self.tmpdir)
+        for package in deps:
+            if not sm.check_installed(package) and not sm.install(package):
+                self.error(package + ' is needed for the test to be run')
+        url = 'https://github.com/bminor/glibc/archive/master.zip'
+        tarball = self.fetch_asset("glibc.zip", locations=[url], expire='0d')
+        archive.extract(tarball, self.srcdir)
+        self.srcdir = os.path.join(self.srcdir, "glibc-master")
+        os.chdir(self.build_dir)
+        process.run(self.srcdir + '/configure --prefix=%s' % self.build_dir,
```

Comment:
if you do not want to retain `os.chdir(self.build_dir) ` please use `os.path.join() to add dir and file`